src/moments.jl

```diff
@@ -85,14 +85,16 @@ end
 function varm(A::RealArray, w::AbstractWeights, M::RealArray, dim::Int;
                    corrected::DepBool=nothing)
     corrected = depcheck(:varm, corrected)
-    varm!(similar(A, Float64, Base.reduced_indices(axes(A), dim)), A, w, M,
+    T = promote_type(eltype(A), eltype(w), eltype(M))
```

Comment:
Likewise, here and below it should be possible to use `zero` to choose the appropriate type? That's not supported currently given the signature, but that would allow e.g. computing the standard deviation of Unitful types.